### DISCUSSION TITLE
OpenSSL speed - unknown algorithm

### DISCUSSION THREAD

**Discussion by dalibor-rada** [open-quantum-safe/oqs-provider#384] [Q&A] [answered]
Hi,

I would like to make a benchmark for postquantum algorithms in OpenSSL via the openssl speed command. I installed liboqs, then I managed to make running the oqs-provider aswell. Then i added this provider to the openssl.cnf file, so the list of providers looks like this
**$openssl list -providers**
providers:
  default
    name: OpenSSL Default Provider
    version: 3.0.2
    status: active
  oqsprovider
    name: OpenSSL OQS Provider
    version: 0.5.3
    status: active

But when I try to run the openssl speed command, it says Unknown algorithm <name>. I followed the instructions from this github. Any ideas how to fix this problem, please?

I look forward to any response :)

**Comment by baentsch** [comment]
> Any ideas how to fix this problem, please?

The above does not contain sufficient information to allow us to help. Please submit a full [bug report](https://github.com/open-quantum-safe/oqs-provider/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=).

Please also consider reading the code in [profiling](https://github.com/open-quantum-safe/profiling) which already contains a feature to do exactly what you want to achieve.

**Reply by dalibor-rada** [reply]
Hello,

thanks for quick response. I did sumbit a full bug report, hopefully it will help.

And about profiling, is there some type of tutorial or something to make it running? It would be helpful to me, because the building via dokcer does not work to me.

**Comment by baentsch** [comment, marked as answer]
As documented [here](https://github.com/open-quantum-safe/oqs-provider/issues/385#issuecomment-2041042292) the problem was a [downlevel openssl version](https://github.com/open-quantum-safe/oqs-provider/?tab=readme-ov-file#note-on-openssl-versions).
